pkgs/development/libraries/libusbmuxd/default.nix

```diff
@@ -1,24 +1,44 @@
-{ lib, stdenv, fetchFromGitHub, autoreconfHook, pkg-config, libplist }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoreconfHook
+, pkg-config
+, libplist
+, libimobiledevice-glue
+}:
 
 stdenv.mkDerivation rec {
   pname = "libusbmuxd";
-  version = "unstable-2021-02-06";
+  version = "unstable-2022-05-04";
 
   src = fetchFromGitHub {
     owner = "libimobiledevice";
     repo = pname;
-    rev = "3eb50a07bad4c2222e76df93b23a0161922150d1";
-    sha256 = "sha256-pBPBgE6s8JYKJYEV8CcumNki+6jD5r7HzQ0nZ8yQLdM=";
+    rev = "36ffb7ab6e2a7e33bd1b56398a88895b7b8c615a";
+    hash = "sha256-41N5cSLAiPJ9FjdnCQnMvPu9/qhI3Je/M1VmKY+yII4=";
   };
 
-  nativeBuildInputs = [ autoreconfHook pkg-config ];
-  buildInputs = [ libplist ];
+  # write version including last tagged release so downstream package version
+  # checks will pass
+  postPatch = ''
+    echo '2.0.2-${version}' > .tarball-version
```

Comment:
Sure, that version format sounds like a good idea. I'll have to double check that `pkg-config` is happy parsing it.

I know nothing about update scripts, do you have any pointers? Does having an update script imply that the package will be automatically updated? That seems a little dangerous for these tools.

Reply:
If pkg-config does not like it, you can always post-process it. The important thing is that the base number will be kept up to date.

See https://nixos.org/manual/nixpkgs/stable/#var-passthru-updateScript and https://github.com/NixOS/nixpkgs
/blob/d9c647d7d8b3c1989c28cbefdfca6680f820bd23/maintainers/scripts/update.nix#L139-L171.

Yes, nixpkgs-update bot is not currently able to update this package because it will try to update the version attribute, which won’t change the `src` attribute. Adding an update script will allow it to open PRs for you the package maintainers to review. (Though that can be disabled in <https://github.com/ryantm/nixpkgs-update/blob/934b497d8e32a42238a9525126614ad74142da78/src/Skiplist.hs>.)

Reply:
I changed all the package versions to the RFC107 format and `pkg-config` is happy with it.

I declined to add an update script because these packages need to be upgraded together and really should be subject to hardware testing. I'll leave the decision to the package maintainers.

Reply:
Even then, the update script will make it easier than manually updating base version, commit hash and output hash. And nixpkgs-update is going to gain support for updating packages from git eventually, so you will need to blacklist these either way.